src/components/discover-sheet/DiscoverSheet.js

```diff
@@ -15,27 +15,43 @@ import {
 } from 'react-native-yet-another-bottom-sheet';
 const renderHeader = yPosition => <DiscoverSheetHeader yPosition={yPosition} />;
 
+function useAreHeaderButtonVisible() {
+  const [isSearchModeEnabled, setIsSearchModeEnabled] = useState(true);
+  return [{ isSearchModeEnabled, setIsSearchModeEnabled }, isSearchModeEnabled];
+}
+
 function DiscoverSheetAndroid() {
+  const [headerButtonsHandlers, deps] = useAreHeaderButtonVisible();
+
+  const value = useMemo(
+    () => ({
+      ...headerButtonsHandlers,
+    }),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    [deps]
+  );
   return (
-    <YABSForm
-      panGHProps={{
-        maxPointers: 17, // magic number for duck typing on native side
-        simultaneousHandlers: 'AnimatedScrollViewPager',
-      }}
-      points={[0, 200, deviceUtils.dimensions.height - 200]}
-      style={[
-        StyleSheet.absoluteFillObject,
-        {
-          backgroundColor: 'white',
-          bottom: 0,
-          top: 100,
-        },
-      ]}
-    >
-      <YABSScrollView style={{ flex: 1, height: '100%' }}>
-        <DiscoverSheetContent />
-      </YABSScrollView>
-    </YABSForm>
+    <DiscoverSheetContext.Provider value={value}>
```

Comment:
Looks like al the props of the children components could be declared outside of the render method (`style`, `points`, `panGHProps`)

Reply:
yeah, but at the same time android will be totally reinvented so lemme leave it like this for now